emlog.h

```diff
@@ -39,6 +39,7 @@ struct emlog_info {
     size_t write_point;         /* Offset in circ. buffer of newest data */
     loff_t offset;              /* Byte number of read_point in the stream */
     struct emlog_info *next;
+    rwlock_t rwlock;            /* per e-info rwlock */
```

Comment:
Maybe mention additionally that it's for protection from concurrent readers/writers?